### PR TITLE
[RFC] log.c;message.c;strings.c add func_attr_printf

### PR DESCRIPTION
**Pull request by MichaHoffmann**
To add func_attr_printf to various functions and fix the resulting compiler warnings.

### CODE REVIEW

File: src/nvim/buffer_updates.c [hunk 1]
```diff
@@ -180,7 +180,7 @@ void buf_updates_send_changes(buf_T *buf,
   // change notifications are so frequent that many dead channels will be
   // cleared up quickly.
   if (badchannelid != 0) {
-    ELOG("Disabling buffer updates for dead channel %llu", badchannelid);
+    ELOG("Disabling buffer updates for dead channel %lu", badchannelid);
```
Comment (justinmk) [inline src/nvim/buffer_updates.c]: I think we normally use `%zu` for `uint64_t` (and this is compatible with MSVC)

Comment (MichaHoffmann) [reply src/nvim/buffer_updates.c]: Okay, will change that!

Comment (jamessan) [reply src/nvim/buffer_updates.c]: Yeah, `%zu` would be incorrect on 32-bit platforms.  `"%" PRIu64` is the correct fix.

File: src/nvim/eval.c [hunk 2]
```diff
@@ -1940,10 +1940,10 @@ static char_u *ex_let_one(char_u *arg, typval_T *const tv,
   } else if (*arg == '@') {
     arg++;
     if (op != NULL && (*op == '+' || *op == '-')) {
-      emsgf(_(e_letwrong), op);
+      EMSG2(_(e_letwrong), op);
```
[the remark this answers is not in the document]
Comment (MichaHoffmann) [reply src/nvim/eval.c]: did not know that!

[the remark this answers is not in the document]
Comment (MichaHoffmann) [reply src/nvim/eval.c]: i'll leave the EMSG(msg) changes because emsg requires a char_u ptr.

**Comment by justinmk** [comment]
Nice, thanks Michael!

We could remove `[I]EMSG{2,3}` entirely, they're useless.

**Pull request merged**